packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -365,57 +375,52 @@ class _DefaultPub implements Pub {
     String? flutterRootOverride,
   }) async {
     int exitCode;
-    int attempts = 0;
-    int duration = 1;
-
-    List<_OutputLine>? output;
-    StreamSubscription<String> recordLines(Stream<List<int>> stream, _OutputStream streamName) {
-      return stream
-        .transform<String>(utf8.decoder)
-        .transform<String>(const LineSplitter())
-        .listen((String line) => output!.add(_OutputLine(line, streamName)));
-    }
+    String lastPubMessage = 'no message';
 
-    final Status? status = printProgress
-      ? _logger.startProgress('Running "flutter pub $command" in ${_fileSystem.path.basename(directory)}...',)
-      : null;
     final List<String> pubCommand = _pubCommand(arguments);
     final Map<String, String> pubEnvironment = await _createPubEnvironment(context, flutterRootOverride);
+    pubEnvironment[_kPubForceTerminalOutputKey] = '1';
```

Comment:
I guess this should depend on whether flutter_tools itself has a terminal attached.

Reply:
That's a good catch. So, I checked flutter_tools' logging code and couldn't see any place that it makes the same differentiation. I'll have to double check, but it looked like flutter_tools as it is does not change output when there's no terminal attached.

Reply:
There is some logic in the Logger class:

https://github.com/flutter/flutter/blob/8c8bb22222b6739ff98518c8ae9d48bb839b5102/packages/flutter_tools/lib/src/base/logger.dart#L32

Reply:
I guess with the new `inheritStdio` approach, we can forego this check.